azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -657,6 +669,31 @@ String getFlowContainerMemoryRequest(final Map<String, String> flowParam) {
     return userMemoryRequest;
   }
 
+  /**
+   * This method is used to get disk request for flow container. Precedence is defined below.
+   * a) Use disk request set in flow parameter constrained by max allowed disk set in config
+   * b) Use disk request set in system properties or default is set in @diskRequest
+   *
+   * @param flowParam
+   * @return Disk request for a flow container
+   */
+  @VisibleForTesting
+  String getFlowContainerDiskRequest(final Map<String, String> flowParam) {
+    if (flowParam == null || flowParam.isEmpty() || !flowParam
+        .containsKey(FlowParameters.FLOW_PARAM_FLOW_CONTAINER_DISK_REQUEST)) {
+      return this.diskRequest;
+    }
+
+    String userDiskRequest =
+        flowParam.get(FlowParameters.FLOW_PARAM_FLOW_CONTAINER_DISK_REQUEST);
+    int resourceCompare = compareResources(this.maxAllowedDisk, userDiskRequest);
+    if (resourceCompare < 0) {
```

Comment:
Based on the documentation of `compareResources` method it looks like zero will be return if there is an issue with the comparison. Shouldn't we use `<= 0` in that case?
Also if the requested storage is bigger that max allowed shouldn't we default to max instead of min to user match request as much as possible?

Reply:
Let me fix the logic. Good catch.